nmostesting/suites/IS1101Test.py

```diff
@@ -303,585 +247,236 @@ def test_01_03(self, test):
+        for inputId in self.is11_utils.sampled_list(self.base_edid_inputs):
+            sender_ids = self.get_inputs_senders(test, inputId)
 
-                valid, response = TestHelper.do_request(
-                    "GET", self.node_url + "devices/" + input["device_id"]
-                )
-                if not valid:
+            in_version_1 = ""
+            in_version_2 = ""
+            in_version_3 = ""
+
+            snd_versions_1 = {}
+            snd_versions_2 = {}
+            snd_versions_3 = {}
+
+            for sender_id in sender_ids:
+                snd_versions_1[sender_id] = ""
+                snd_versions_2[sender_id] = ""
+                snd_versions_3[sender_id] = ""
+
+            valid, response = self.do_request("GET", self.compat_url + "inputs/" + inputId + "/properties")
+            if not valid or response.status_code != 200:
+                return test.FAIL("Unexpected response from "
+                                 "the Stream Compatibility Management API: {}".format(response))
+            try:
+                in_version_1 = response.json()["version"]
+            except json.JSONDecodeError:
+                return test.FAIL("Non-JSON response returned from the Stream Compatibility Management API")
+            except KeyError as e:
+                return test.FAIL("Unable to find expected key: {}".format(e))
+
+            for sender_id in sender_ids:
+                valid, response = self.do_request("GET", self.node_url + "senders/" + sender_id)
+                if not valid or response.status_code != 200:
                     return test.FAIL("Unexpected response from the Node API: {}".format(response))
-                if response.status_code != 200:
-                    return test.FAIL("The device {} Node API request has failed: {}"
-                                     .format(input["device_id"], response.json()))
                 try:
-                    device = response.json()
-                    if device["id"] != input["device_id"]:
-                        return test.FAIL("device_id does'nt match.")
+                    snd_versions_1[sender_id] = response.json()["version"]
                 except json.JSONDecodeError:
-                    return test.FAIL("Non-JSON response returned from Node API")
+                    return test.FAIL("Non-JSON response returned from the Node API")
                 except KeyError as e:
                     return test.FAIL("Unable to find expected key: {}".format(e))
 
-                self.version[device["id"]] = input["version"]
+            # PUT the Base EDID to the Input
+            valid, response = self.do_request("PUT",
+                                              self.compat_url + "inputs/" + inputId + "/edid/base",
+                                              headers={"Content-Type": "application/octet-stream"},
+                                              data=VALID_EDID)
```

Comment:
Add reasonable delay after Put command.
`start_time = time.time()`
`            while time.time() < start_time + CONFIG.API_PROCESSING_TIMEOUT:`
`                time.sleep(0.2)`

Reply:
The version of an Input should increment immediately after a PUT, isn't it?